### ISSUE TITLE
`task:get_from([type])` returns `nil` instead of a table if `@` is missing

### ISSUE THREAD

**Issue by moisseev** [vstakhov/rspamd#1288]
### Classification (Please choose *one* option):

* [ ] Crash/Hang/Data loss
* [ ] WebUI/Usability
* [ ] Serious bug
* [ ] Other bug
* [ ] Feature
* [x] Enhancement

### Reproducibility (Please choose *one* option):

* [X] Always
* [ ] Sometimes
* [ ] Rarely
* [ ] Unable
* [ ] I didn’t try
* [ ] Not applicable

### Rspamd version:
1.5.0-git.b9bce69
### Description (Please provide a descriptive summary of the issue):
`task:get_from([type])` returns `nil` instead of a table if `@` is missing in the address.
I suggest to return a table with `nil`s for undetectable parts so available information could be retrieved.

### Steps to reproduce:
```
# rspamc
From: name <addr>
MIME-Version: 1.0
Content-Type: multipart/alternative;

```
### Expected results:
`from={[1] = {[addr] = addr, [user] = nil, [name] = name, [domain] = nil}}`
### Actual results:
`from={}, from[1]=nil`

**Comment by vstakhov** [comment]
That doesn't conform RFC I'm afraid. Mailbox definition is the following:
 
```
Mailbox        = Local_part "@" (address_literal | Domain);
```

Hence, addr with no `@` is not valid.

**Comment by moisseev** [comment]
After baa008ed3bb6f0f31783478d8069e5e8fc5ae7d4 it returns wrong `[addr]` value:
`from={[1] = {[addr] = add, [user] = , [name] = name, [domain] = }}`

**Comment by moisseev** [comment]
Resolved with baa008ed3bb6f0f31783478d8069e5e8fc5ae7d4, f93080e713b886f313bf31bd51dd9c397cec8cb1
